news/2024-04-30-clang-everywhere.md

```diff
@@ -0,0 +1,39 @@
+# Clang now available as compiler for all platforms
+
+Our compiler stack per platform generally uses the "default" compiler for that
+platform, see e.g. [here](https://conda-forge.org/docs/maintainer/infrastructure/#compilers-and-runtimes).
+
+In practice, this meant that
+
+```yaml
+c_compiler:
+  - gcc         # [linux]
+  - clang       # [osx]
+  - vs2019      # [win]
+cxx_compiler:
+  - gxx         # [linux]
+  - clangxx     # [osx]
+  - vs2019      # [win]
+```
+
+was the only possible choice for C/C++ compilers.
+
+Recently, we finished adding preliminary support `clang` / `clangxx` as
+C/C++ compilers also on osx and windows, starting from clang 18.
+This is still very fresh, so bugs are possible, and we ask not to change
+the default compilers on feedstocks unless there are compelling reasons.
+
+In any case, it is now possible to use the following configuration in
+`recipe/conda_build_config.yaml`:
+
+```yaml
+# please consult @conda-forge/core before doing this
```

Comment:
Why?

Reply:
My thinking was that we don't (yet?) want to make it a free-for-all for feedstocks to choose their compilers. I think the cautious choice would be to keep feedstocks on the default compilers unless there are good reasons for choosing clang, at least while we get some more experience with using clang on linux/win.

That line was meant as a kind of natural "brake" on people's enthusiasm in order to have a softer rollout, but if you feel that this caution is not necessary, I can remove that line.

Reply:
My feeling is that people that choose `clang` as compiler either know what they are doing and thus don't need to spam `core` with more notifications or they are doing it as part of a copy/paste from another recipe, fail to apply it and will ping someone competent anyways. Thus, for the sake of a bit less notifications, I would omit it.

Reply:
@isuruf, I'm assuming your concern here is addressed now?